ee/secureenclaverunner/secureenclaverunner.go

```diff
@@ -0,0 +1,305 @@
+//go:build darwin
+// +build darwin
+
+package secureenclaverunner
+
+import (
+	"context"
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"os/user"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/ee/consoleuser"
+	"github.com/kolide/launcher/pkg/backoff"
+	"github.com/kolide/launcher/pkg/traces"
+)
+
+const (
+	publicEccDataKey = "publicEccData"
+)
+
+type secureEnclaveRunner struct {
+	uidPubKeyMap        map[string]*ecdsa.PublicKey
+	secureEnclaveClient secureEnclaveClient
+	store               types.GetterSetterDeleter
+	slogger             *slog.Logger
+	mux                 *sync.Mutex
+	interrupt           chan struct{}
+	interrupted         atomic.Bool
```

Comment:
Do you think it's worth updating the other rungroup actors to use atomic.Bool if they use the `interrupt`/`interrupted` pattern? Let me know if yes -- I can file an issue to tackle that in the future 🙂

Reply:
yea, I think so, feels more correct

Reply:
Done! https://github.com/kolide/launcher/issues/2010